experiments/conductor/cmd/runner/runner.go

```diff
@@ -835,3 +861,78 @@ func setSkipOnBranchModifier(opts *RunnerOptions, branch Branch, workDir string)
 	}
 	return branch
 }
+
+func diffLastNCommits(opts *RunnerOptions, branch Branch) {
+	close := setLoggingWriter(opts, branch)
+	defer close()
+	workDir := opts.branchRepoDir
+
+	ctx := context.TODO()
+	checkoutBranch(ctx, branch, workDir)
+
+	// Run git diff command
+	cfg := CommandConfig{
+		Name: "Git diff",
+		Cmd:  "git",
+		Args: []string{
+			"diff",
+			"-r",
+			fmt.Sprintf("HEAD~%d", opts.numCommits),
+		},
+		WorkDir:    workDir,
+		MaxRetries: 1,
+	}
+	output, _, err := executeCommand(opts, cfg)
+	if err != nil {
+		log.Printf("Git diff error for branch %s: %v", branch.Name, err)
+		return
+	}
+
+	// Print the diff output
+	log.Printf("Diff for branch %s (last %d commits):\n%s", branch.Name, opts.numCommits, output)
+	fmt.Printf("Diff for branch %s (last %d commits):\n%s", branch.Name, opts.numCommits, output)
+}
+
+func revertLastNCommits(opts *RunnerOptions, branch Branch) {
```

Comment:
This could mess up the working dir easily. Can we add a condition check that the N-revert should not include "Merged" commits.

Reply:
I print the diff of whats being reverted and ask confirmation from the user before reverting. This is done for each branch. 

How do i check for merged commit ? What git command ?

Reply:
Yeah, the interactive mode is nit but then we cannot run it in the background. You can use `git log --merges`, like `git log  --name-only --oneline --merges`. Or if done in code, I prefer checking the # of parents (merge commit has at least 2)

Reply:
I see.. this command is not meant to be run in background. Will see how to add the merge check